### ISSUE TITLE
Please keep current table context even after connection is disconnected

### ISSUE THREAD

**Issue by jahlbornG** [dbeaver/dbeaver#35536]
**Is your feature request related to a problem? Please describe.**
It is very frustrating that the table context is lost whenever the connection to the db is closed.  i will typically be working with a table for a while, using different filters at the top related to some situation i am investigating.  while the connection remains open to the db, i can switch back and forth to previous filters i have used, etc.  however, when i move on to some other work and the database connection times out, or i stop working for the day and the connection is closed, when i come back, all that rich history is lost.  if i was looking at specific records with specific ids, all that info is lost, and i need to go track down which ids i was looking at using an external source so that i can get back to where i was previously.

**Describe the solution you'd like**
It would be ideal if open table tabs maintained their filter history even when the database is disconnected, re-connected.  it's fine if that information is lost if i close the tab, but if the tab remains open, i would expect that history to remain.

**Comment by ivgag** [comment]
It's not quite clear what you mean under the history. 
We can not save specific IDs and result sets because the data could change and DBeave strive to show the actual data, but you can save your filters. 
![Image](https://github.com/user-attachments/assets/5dfe0722-f129-4e42-8df1-2b7a0215b04b)

**Comment by jahlbornG** [comment]
> It's not quite clear what you mean under the history. We can not save specific IDs and result sets because the data could change and DBeave strive to show the actual data, but you can save your filters. ![Image](https://private-user-images.githubusercontent.com/17929047/373266517-5dfe0722-f129-4e42-8df1-2b7a0215b04b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg1MDQzNDQsIm5iZiI6MTcyODUwNDA0NCwicGF0aCI6Ii8xNzkyOTA0Ny8zNzMyNjY1MTctNWRmZTA3MjItZjEyOS00ZTQyLThkZjEtMmI3YTAyMTViMDRiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDA5VDIwMDA0NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTRlYjMwZTdhNDNlY2JmNjc4OTgzNGMwOTA1M2NjMDRiY2VlMWUxYjgxZDI2MGQ4NzA5ZjRkNTE1MzczZTI4NTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.HdNtxQRVY0vLEzeRfFRlVtT0PqQvZpGyGlMi_72r3w4)

@ivgag  thanks for your response.  no, i don't mean the results because those could absolutely change.  what i mean is the sql text that is currently in the filter section (maybe i'm using the wrong term?  the box that is showing "Enter a Sql expression to filter..." in your image) and the history which is accessible via the right and left green arrows which are to the right of that filter section.

i.e. i'm investigating something in production and i have "uid = 123345" as my current filter because i'm looking at that row (and i previously used "uid = 9876", so that is available under the left green arrow).  when i disconnect the db and reconnect the next day, i'd like to have those filters (or whatever you might call them) still available.  i don't want to save these as a long term filter or anything because they won't have any meaning in a few days when i am no longer looking into the current issue.  i just want my "current working state" to be maintained.

to add some more clarification, i'd be perfectly fine if the info were only maintained while the tab was open.  i.e. if i close the tab and then re-open the table (creating a new tab), i'm fine if i start with a blank slate.  but while i keep the tab open, i'd like to not lose the current context of what i was doing.

**Comment by arhayka** [comment]
Thank you for the idea!
We could try to keep filters when disconnecting.
We wonder how many users seek this feature.
So, for someone who is interested, please vote for the issue with 👍 under the issue description to promote it to a higher priority.

**Comment by jahlbornG** [comment]
@arhayka i know this is waiting for votes.  i'm not sure how much traffic these features get on github.  it's very frustrating to lose the context just because the db connection is closed.  any chance this could be moved forward without getting votes?  i'm pretty sure this is something that people would appreciate once they got it.

**Comment by sashsvamir** [comment]
I've been waiting for this feature for a long time. I wrote an issue, but it was closed because it duplicated current issue.
I'm still waiting

**Comment by E1izabeth** [comment]
We have implemented history for the filters. Now you can see them here:

<img width="1067" height="206" alt="Image" src="https://github.com/user-attachments/assets/840c9a55-e156-4969-a80a-c4f03a2715c1" />

So, if your connection has been reestablished, you can choose the latest filter from the list.

Please try this feature in DBeaver 26.1.0 and let me know if it satisfies your needs.

**Comment by sashsvamir** [comment]
@E1izabeth I tested this version [dbeaver-ce-26.1.1-macos-aarch64.dmg](https://dbeaver.io/files/ea/dbeaver-ce-26.1.1-macos-aarch64.dmg)
But I didn't see the filter string saved like in your screenshot. It's always empty (after reconnection), even though I entered the SQL string for the filter.
Maybe I need to activate something in the settings?

**Comment by E1izabeth** [comment]
Hmm.. Have you applied this filter? Or you're just typing and then get disconnected and loose it?

**Comment by sashsvamir** [comment]
Yes, I entered a SQL query, filtered the list by it (pressed enter), then reconnected the network, pressed reconnect in the app, and after reconnecting, I see an empty SQL string, and its drop-down menu does not contain any saves, as in your screenshot.

**Comment by E1izabeth** [comment]
Ah, I see. It's only a PRO edition's feature.
Thanks

**Comment by sashsvamir** [comment]
@E1izabeth Is this version available for free? How do I download it?

**Comment by E1izabeth** [comment]
@sashsvamir 
The feature I was talking about will be available in DBeaver 26.1.0 in PRO versions. It's a paid version. Release is planned for June 8th. However, a free two-week trial will be available: https://dbeaver.com/trial/

Also, you can join the Early Access Program now and use it for free https://dbeaver.com/docs/dbeaver/Early-Access-Program-license/
